libs/util/kafka/src/lib/serializer/json/KafkaMessageJsonSerializer.ts

```diff
@@ -0,0 +1,84 @@
+import type {
+  DecodedKafkaMessage,
+  IKafkaMessageSerializer,
+  Json,
+  KafkaMessage,
+} from "../../../types";
+
+class KafkaMessageJsonSerializer implements IKafkaMessageSerializer {
+  private deserializeField(field: Buffer | null): string | Json | undefined {
+    if (field === undefined || field === null) {
+      return undefined;
+    }
+    if (!Buffer.isBuffer(field)) {
+      return field;
+    }
+
+    // A field with the "leading zero byte" indicates the schema payload.
+    // The "content" is possibly binary and should not be touched & parsed.
+    if (
+      Buffer.isBuffer(field) &&
+      field.length > 0 &&
+      field.readUInt8(0) === 0
+    ) {
+      throw new Error(
+        "Field failed to deserialize. A field with the leading zero byte indicates the schema payload."
+      );
+    }
+
+    let result = field.toString("utf8");
+    const startChar = result.charAt(0);
+
+    // only try to parse objects and arrays
+    if (startChar === "{" || startChar === "[") {
+      try {
+        result = JSON.parse(field.toString());
+      } catch (e) {
+        /* empty */
```

Comment:
Can we add logs or throw here? its a problem to leave empty catch

Reply:
It's actually intended to catch this and return the original object. I'll add a warning.